### PR TITLE
Add initial rust frontend

### PR DESCRIPTION
**Pull request by ohaz**

### CODE REVIEW

**Comment by mxwnk** [comment]
Minor: we did not use the framework / language name for the other projects, so `frontend-desktop` would be fine for me instead of `frontend-desktop-rust`

**Comment by ohaz** [comment]
> Minor: we did not use the framework / language name for the other projects, so `frontend-desktop` would be fine for me instead of `frontend-desktop-rust`

Will we ever have multiple desktop frontends? But yeah, we can rename it to `-rust` when the time comes.

File: frontend-desktop/src/main.rs [hunk 1]
```diff
@@ -22,27 +38,38 @@ fn wsconnect() -> Client{
...
     ]));
 
+
     let main_window_weak = ui.as_weak();
+    let history_model_clone = history_model.clone();
+    timer.start(TimerMode::Repeated, std::time::Duration::from_millis(2), move || {
+        let mut guard: MutexGuard<'_, Option<JoinMessage>> = mutex.lock().unwrap();
+        if let Some(item) = &*guard {
+            let history_item: HistoryItem = HistoryItem { text: "wants to go for a coffee".into(), user: item.name.clone().into() };
+            history_model_clone.push(history_item);
```
Comment (t-animal) [inline frontend-desktop/src/main.rs]: Why does this show up in the UI? I thought we were working on a clone.

**Review by t-animal — APPROVED** [review]

File: frontend-desktop/src/main.rs (hunk 1 above)
Comment (ohaz) [reply frontend-desktop/src/main.rs]: Thought so too but it seems to work. Need someone with more rust knowledge

**Review by mxwnk — APPROVED** [review]

**Pull request merged**